File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

n = int(input())
stack = list()
result = list()
cur = 1

for _ in range(n):
    data = int(sys.stdin.readline())
    while cur <= data:
        stack.append(cur)
        result.append('+')
        cur += 1
    if stack[-1] == data:
        stack.pop()
        result.append('-')
    else:
        print("NO")
        exit(0)

else:
    for i in result:
        print(i)

File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());

        Stack<Integer> stack = new Stack<>();
        StringBuilder result = new StringBuilder();
        int currentNumber = 1;

        for (int i = 0; i < n; i++) {
            int number = Integer.parseInt(br.readLine());
            while (currentNumber <= number) {
                stack.push(currentNumber);
                result.append("+").append("\n");
                currentNumber += 1;
            }

            if (stack.peek() == number) {
                stack.pop();
                result.append("-").append("\n");
            } else {
                result.setLength(0); // StringBuilder 초기화
                result.append("NO");
                break;
            }
        }
        bw.write(result.toString());
        br.close();
        bw.close();
    }
}


